src/module/actor/lancer-actor.ts

```diff
@@ -1119,19 +1136,22 @@ export class LancerActor<T extends LancerActorType> extends Actor {
         return "1";
     }
   }
+
+  // Typeguards
+  is_pilot(): this is LancerActor & { data: { type: EntryType.PILOT } } {
+    return this.data.type === EntryType.PILOT;
+  }
+  is_mech(): this is LancerActor & { data: { type: EntryType.MECH } } {
+    return this.data.type === EntryType.MECH;
+  }
+  is_npc(): this is LancerActor & { data: { type: EntryType.NPC } } {
+    return this.data.type === EntryType.NPC;
+  }
+  is_deployable(): this is LancerActor & { data: { type: EntryType.DEPLOYABLE } } {
+    return this.data.type === EntryType.DEPLOYABLE;
+  }
```

Comment:
I still think these type guards should state the narrower types `this is LancerActor & { data: LancerActorDataProperties<EntryType.PILOT> }` and friends.

Reply:
If this is that important to you you can write the patch yourself, but as I've already explained, intersecting with that subset of `Actor.data` is no more specific than the subset I'm using here.

Reply:
And as I've explained, what types are narrower/more specific than others is a property that is independent of what a type environment infers. If you want to make me write the patch, I will; this just seems like a weird hill to die on.

Reply:
I'm not continuing this conversation.